src/Servers.jl

```diff
@@ -254,30 +257,34 @@ function listenloop(h, server,
                 close(io)
                 continue
             end
-            connectioncounter[] += 1
-            conn = Connection(server.hostname, server.hostport, 0, 0, true, io)
+            connection_count[] += 1
+            conn = Connection(io)
+            conn.host, conn.port = server.hostname, server.hostport
             let io=io, count=count
-                @async begin
-                    try
-                        verbose && @info "Accept ($count):  $conn"
-                        handle(h, conn, reuse_limit, readtimeout)
-                        verbose && @info "Closed ($count):  $conn"
-                    catch e
+                @async try
+                    verbose && @info "Accept ($count):  $conn"
+                    handle_connection(h, conn, reuse_limit, readtimeout)
+                    verbose && @info "Closed ($count):  $conn"
+                catch e
+                    if e isa Base.IOError && e.code == -54
+                        verbose && @warn "connection reset by peer (ECONNRESET)"
+                    else
                         @error exception=(e, stacktrace(catch_backtrace()))
```

Comment:
We should rethrow here, at least for some error types.
But, we need a way to get these exceptions into the main task. An enclosing `@sync` should be enough?
We should also add test cases that `@assert false` in each of the transaction, connection and listen tasks and ensure that all are rethrown by the listen task (the main task).

Reply:
Opened up an issue to track the suggested changes: https://github.com/JuliaWeb/HTTP.jl/issues/324